### PR TITLE
Added contributing file

### PR DESCRIPTION
**Pull request by leiamcalisteryoung**
Write contributing file according to issue #394

### CODE REVIEW

**Comment by leiamcalisteryoung** [comment]
We've written the first draft of the contributing file. We weren't sure what to say for coding style, so if there's a suggestion for that, let us know.

**Comment by ozgurakgun** [comment]
I guess you can say we use `rustfmt`

**Comment by ozgurakgun** [comment]
This is a great start, thanks. I would add information about the CI (github actions) and how we strive to keep code and documentation coverage high. But good otherwise. Anything else we are forgetting @lixitrixi @niklasdewally @gskorokhod ?

**Review by ozgurakgun — COMMENTED** [review]
This is a great start, thanks. I would add information about the CI (github actions) and how we strive to keep code and documentation coverage high. But good otherwise. Anything else we are forgetting @lixitrixi @niklasdewally @gskorokhod ?

**Comment by niklasdewally** [comment]
> This is a great start, thanks. I would add information about the CI (github actions) and how we strive to keep code and documentation coverage high. But good otherwise. Anything else we are forgetting @lixitrixi @niklasdewally @gskorokhod ?

In general, we might want to remove this stuff from the wiki and just keep it in one place (this file).

Also, we could make a bug report issue template ?

**Comment by niklasdewally** [comment]
Even more tangentially, having bug! output the git version/sha of conjure oxide being used would be sensible!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - Be specific!
+  - Give sample code if you can. [My stackoverflow question](http://stackoverflow.com/q/12488905/180626) includes sample code that *anyone* with a base R setup can run to reproduce what I was seeing
+- What you expected would happen
+- What actually happens
+- Notes (possibly including why you think this might be happening, or stuff you tried that didn't work)
+
+People *love* thorough bug reports. I'm not even kidding.
+
+## Coding Style
+Please write clear and well doccumented code.
```
Comment (gskorokhod) [inline CONTRIBUTING.md:52]: We can add here:

- Run `cargo fmt` in the project directory to automatically format code
- Use `cargo clippy` to lint the code and identify any common issues

**Comment by ozgurakgun** [comment]
> Even more tangentially, having bug! output the git version/sha of conjure oxide being used would be sensible!

Yep, conjure does this. Should be captured as an issue.

**Comment by niklasdewally** [comment]
> > Even more tangentially, having bug! output the git version/sha of conjure oxide being used would be sensible!
> 
> Yep, conjure does this. Should be captured as an issue.

https://github.com/conjure-cp/conjure-oxide/issues/404

**Comment by niklasdewally** [comment]
Closes: #394

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -49,7 +49,8 @@ We use GitHub issues to track public bugs. Report a bug by [opening a new issue]
 People *love* thorough bug reports. I'm not even kidding.
 
 ## Coding Style
-Please write clear and well doccumented code.
+- Run cargo fmt in the project directory to automatically format code
```
Comment (ozgurakgun) [inline CONTRIBUTING.md:52]: very small comment: you can use backticks to format text as code. so maybe `cargo fmt` and `cargo clippy` here.

Comment (ozgurakgun) [reply CONTRIBUTING.md:52]: but I'll merge as is, we can take another pass later as this is not crucial. thanks!

**Review by ozgurakgun — APPROVED** [review]

**Pull request merged**